### DISCUSSION TITLE
How to debug a nuxt-related HUGE memory leak?

### DISCUSSION THREAD

**Discussion by MosheL** [nuxt/nuxt#23087] [Questions]
Vue 3.6 / 3.7 production build. 
Every request with full SSR create a 1.2MB  memory leak in server.

I tried to take a snapshot ( allocation and heap) and saw the routes stored in the memory again and again and did not released after the request gone from server. 

I am trying to understand why the memory did not release but can't know from where to start.

example memory snapshots:
![image](https://github.com/nuxt/nuxt/assets/2265043/18954516-2cf1-4072-bbcd-1ce0541ca8d9)

![image](https://github.com/nuxt/nuxt/assets/2265043/869883ce-f044-4763-a2a2-0250cf65ee7f)

what I  tried:
- Disable SSR - help
- Check global refrences in my code -  fixed (as I know).
- Destroy `__VUE_INSTANCE_SETTERS__`  - not released the memory but broke the app.
- Check ` globalThis.__unctx__.get('nuxt-app').use().$router ` in server after the request finished - it works, but it has the last request nuxt & router context. so the old requests (all except the last one) memory not used from here.

BTW: why in SSR the vue router routes table can't be reused between requests?

Thanks for all!

**Comment by memic84** [comment]
Hey @MosheL 

Did you figure this out? We are experiencing something similar since 3.6 versions. Seeing a lot of matcher / addRoute function being held on from the vue-router.

**Reply by MosheL** [reply]
no, but I was added a workaround - run two instances of the app and recycle between them when they got to specific memory usage, and added a cache to make the problem smaller. ugly, but works.

pm2 can do it, IIS/IISNode can also do it in the same way.

**Reply by memic84** [reply]
Yeah that work too and gives you some time to find the real issue. Have you been able to create a minimal reproduction?
We are running 3.5.1 where the problem doesn't persist, i believe that you are seeing the same? Memory leaks from the version 3.6.x and up?

**Reply by MosheL** [reply]
In our code was a  mem leak in the old version. when I was tried to investigate the  3.6+ problem I  found (and fixed) the internal  mem leak so I can't  compare right now the 3.5 and the 3.6 one. 

other change I was done: make the route table smaller. so the leak is about a ~300KB/req and not the 1.2MB/req so its better.
